### PR TITLE
YONK-876: Remove Image API added for mobile app themes

### PR DESCRIPTION
**Pull request by aamishbaloch**
Changes include addition of API which can remove image (logo or header) of any theme of the specified mobile app. Unit tests also included.

### CODE REVIEW

File: mobileapps/views.py [hunk 1]
```diff
@@ -1129,3 +1129,37 @@ def delete(self, request, theme_id):
...
+
+    def __init__(self):
+        self.permission_classes += (IsStaffOrReadOnlyView,)
+
+    @transaction.atomic
+    def delete(self, request, theme_id, attribute):
+        theme = get_object_or_404(Theme, pk=theme_id)
+
+        if attribute == 'logo_image':
+            theme.logo_image_uploaded_at = None
```
Comment (ziafazal) [inline mobileapps/views.py]: can we delete the image also?

File: mobileapps/views.py [hunk 2]
```diff
@@ -1129,3 +1129,37 @@ def delete(self, request, theme_id):
...
+        - attribute
+            - Logo Image: logo_image
+            - Header Background Image: header_bg_image
+    """
+
+    serializer_class = ThemeSerializer
+    lookup_url_kwarg = 'theme_id'
+
+    def __init__(self):
+        self.permission_classes += (IsStaffOrReadOnlyView,)
```
Comment (ziafazal) [inline mobileapps/views.py]: Can we use a permission witch checks only if request user is staff?. I mean we don't have any readonly view here.

File: mobileapps/tests.py [hunk 3]
```diff
@@ -1612,3 +1612,38 @@ def test_mobileapps_organization_theme_with_no_default_images(self):
...
+
+        response = self.do_delete(reverse(
+            'mobileapps-organization-themes-remove-image', kwargs={
+                'theme_id': organization_theme.id,
+                'attribute': 'header_bg_image'
+            }
+        ))
+        self.assertEqual(response.status_code, 200)
+
+        response = self.do_get(reverse(
```
Comment (ziafazal) [inline mobileapps/tests.py:1641]: Can we add a test which hits image deletion url with invalid `theme_id` and valid attribute and then valid `theme_id` and invalid attribute?

**Review by ziafazal — CHANGES_REQUESTED** [review]
@aamishbaloch completed first round of review.

**Comment by aamishbaloch** [comment]
@ziafazal Done with the changes. Can you review?

File: mobileapps/views.py [hunk 4]
```diff
@@ -1129,3 +1133,41 @@ def delete(self, request, theme_id):
...
+    def __init__(self):
+        self.permission_classes += (IsStaffView,)
+
+    @transaction.atomic
+    def delete(self, request, theme_id, attribute):
+        theme = get_object_or_404(Theme, pk=theme_id)
+
+        if attribute == 'logo_image':
+            theme.remove_logo_image(
+                settings.ORGANIZATION_LOGO_IMAGE_SIZES_MAP,
```
Comment (ziafazal) [inline mobileapps/views.py]: `remove_logo_image` and `remove_header_bg_image` methods encapsulate the data which is being passed via params.

**Review by ziafazal — CHANGES_REQUESTED** [review]
@aamishbaloch I think one change is needed.

**Comment by aamishbaloch** [comment]
@ziafazal Done with the change you suggested. Can you review it now?

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**